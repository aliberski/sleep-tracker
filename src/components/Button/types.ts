import { ViewStyle, TextStyle } from 'react-native';

export interface IProps {
  text: string;
  onPress: () => void;
  testID?: string;
}

export interface IStyle {
  container: ViewStyle;
  text: TextStyle;
}
