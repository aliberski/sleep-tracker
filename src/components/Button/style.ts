import { StyleSheet } from 'react-native';
import {
  COLOR,
  PADDING,
  FONT,
  FONT_SIZE,
} from 'constants/globalStyle';

import { IStyle } from './types';

export default StyleSheet.create<IStyle>({
  container: {
    width: '100%',
    backgroundColor: COLOR.primary,
    paddingHorizontal: PADDING.primary,
    paddingVertical: PADDING.medium,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: COLOR.font_light,
    fontFamily: FONT.bold,
    fontSize: FONT_SIZE.primary,
    textAlign: 'center',
  },

  containerLink: {
    width: '100%',
    padding: PADDING.small,
  },
  textLink: {
    color: COLOR.font,
    fontFamily: FONT.bold,
    fontSize: FONT_SIZE.small,
    textAlign: 'center',
  },
  indicator: {
    position: 'absolute',
    right: PADDING.big,
  },
});
