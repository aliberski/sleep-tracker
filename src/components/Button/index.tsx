import React, { memo } from 'react';
import { TouchableOpacity, Text } from 'react-native';

import style from './style';
import { IProps } from './types';

const Button = (props: IProps) => {
  const { onPress, text } = props;
  return (
    <TouchableOpacity style={style.container} onPress={onPress}>
      <Text style={style.text}>{text}</Text>
    </TouchableOpacity>
  );
};

export { Button };
export default memo(Button);
