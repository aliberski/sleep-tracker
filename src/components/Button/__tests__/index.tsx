import 'react-native';
import React from 'react';

import Button from '../index';
import renderer from 'react-test-renderer';

describe('Button', () => {
  const mockFunction = jest.fn();
  const button = renderer.create(<Button text='123' onPress={mockFunction} />);

  it('renders correctly', () => {
    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
