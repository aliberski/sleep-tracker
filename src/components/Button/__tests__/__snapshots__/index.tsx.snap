// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#6666EE",
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Roboto-Bold",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    123
  </Text>
</View>
`;
